### PR TITLE
Feature/nex 246/delivery creation refactorin

### PR DESCRIPTION
**Pull request by boajer**
TODO:

- ~proper class for created delivery~ 
- ~cleanup dead code~
- ~create LTI container based on~

TODO out of the scope of taks 
- Proper DeliveryContainers for DeliverTenantAPI => add abstraction for LTI credentials => see separate PR

### CODE REVIEW

File: controller/PublishDelivery.php [hunk 1]
```diff
@@ -25,6 +25,7 @@
 use oat\tao\model\taskQueue\Task\TaskInterface;
 use oat\taoDeliverConnect\model\delivery\form\RemoteDeliveryWizardForm;
 use oat\taoDeliverConnect\model\DeliverTenantService;
+use oat\taoDeliverConnect\model\PublicationService;
```
Comment (jbout) [inline controller/PublishDelivery.php]: why?

File: model/tasks/LookupForDeliveryTask.php [hunk 2]
```diff
@@ -85,43 +98,84 @@ public function __invoke($params = [])
...
+
+    /**
+     * @param string $label
+     * @param LookupSnapshot $lookupSnapshot
+     * @param DeliverTenant $deliverTenant
+     * @param RdfClass $deliveryClass
+     * @param string $launchUrl LTI launch URL on Deliver
+     * @return RdfResource
+     */
+    private function createDeliveryObject($label, LookupSnapshot $lookupSnapshot, DeliverTenant $deliverTenant, RdfClass $deliveryClass, $launchUrl)
```
Comment (jbout) [inline model/tasks/LookupForDeliveryTask.php]: Resource?

File: model/PublicationService.php [hunk 3]
```diff
@@ -73,6 +75,18 @@ public function lookup(DeliverTenant $deliverTenant, $publicationId)
...
+    /**
+     * @param DeliverTenant $deliverTenant
+     * @param string $publicationId
+     * @return DeliverDeliverySnapshot
+     * @throws GuzzleException
+     * @throws InconsistentDataException
+     */
+    public function delivery(DeliverTenant $deliverTenant, $publicationId)
+    {
+        return $this->getApiClient()->call(new GetDeliveryCall($deliverTenant, $publicationId));
```
Comment (julien-sebire) [inline model/PublicationService.php]: Isn't this and GetDeliveryCall kind of a duplicate of the LaunchUriCall? Can we merge both? They're making a call to the same url, only the parsing of the response differs, and this one is more complete.

File: model/delivery/DeliverDeliverySnapshot.php [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+    /**
+     * @return string
+     */
+    public function getLabel()
+    {
+        return $this->label;
+    }
+
+
+}
```
Comment (julien-sebire) [inline model/delivery/DeliverDeliverySnapshot.php]: Missing new line at end of file.

File: model/delivery/factory/RemoteDeliveryFactory.php [hunk 5]
```diff
@@ -84,34 +78,12 @@ public function create(
...
-
-        $eventManager->trigger(new DeliveryPublicationCreatedEvent($deliverTenant->getId(), $publicationId));
+        $eventManager = $this->getServiceLocator()->get(EventManager::class);
+        $eventManager->trigger(new DeliveryPublicationCreatedEvent($deliverTenant->getId(), $publicationId, $deliveryClass->getUri()));
 
         return new Report(
             Report::TYPE_SUCCESS,
-            __('Remote delivery successfully created.'),
-            $deliveryResource
+            __('Publication on remove deliver created.')
```
Comment (julien-sebire) [inline model/delivery/factory/RemoteDeliveryFactory.php]: "remove" => "remote"

File: model/PublicationService.php (hunk 3 above)
Comment (boajer) [reply model/PublicationService.php]: TBH I can't find `LaunchUrlCall` implementation, but unittests for that class. So I assume we are ok :)

**Pull request merged**